clients/src/main/java/org/apache/kafka/clients/consumer/internals/Heartbeat.java

```diff
@@ -66,12 +73,16 @@ void sentHeartbeat(long now) {
         heartbeatInFlight = true;
         update(now);
         heartbeatTimer.reset(rebalanceConfig.heartbeatIntervalMs);
+
+        log.trace("Sending heartbeat request with {}ms remaining on timer", heartbeatTimer.remainingMs());
```

Comment:
Should this be inside `isTraceEnabled()` to avoid computing `remainingMs()` in the case that trace logging isn't on?

```suggestion
        if (log.isTraceEnabled()) {
            log.trace("Sending heartbeat request with {}ms remaining on timer", heartbeatTimer.remainingMs());
        }
```

(also below)